packages/amplify-prompts/src/prompter.ts

```diff
@@ -0,0 +1,212 @@
+    });
+
+    actions.ctrl.a = 'a';
+
+    let result = typeof choices[0] === 'string' ? choices[0] : ((choices[0] as GenericChoice<T>).name as string | string[]);
+
+    if (choices?.length === 1) {
+      this.print.info(`Only one option for ${message}. Selecting ${result} by default.`);
+    } else {
+      ({ result } = await this.prompter<{ result: M extends 'many' ? string[] : string }>({
+        // actions is not part of the TS interface but it's part of the JS API
+        // eslint-disable-next-line @typescript-eslint/ban-ts-comment
+        // @ts-ignore
+        actions,
+        // footer is not part of the TS interface but it's part of the JS API
+        // eslint-disable-next-line @typescript-eslint/ban-ts-comment
+        // @ts-ignore
+        footer: opts?.multiselect ? '(Use <space> to select, <ctrl + a> to toggle all)' : undefined,
+        type: 'autocomplete',
+        name: 'result',
+        message,
+        initial: opts?.initial,
+        // there is a typo in the .d.ts file for this field -- muliple -> multiple
+        // eslint-disable-next-line @typescript-eslint/ban-ts-comment
+        // @ts-ignore
+        multiple: opts?.multiselect,
+        choices: enquirerChoices,
+      }));
+    }
+
+    if (Array.isArray(result)) {
+      return result.map(item => choiceValueMap.get(item) as T) as PickReturn<M, T>;
+    } else {
+      // result is a string
+      return choiceValueMap.get(result as string) as PickReturn<M, T>;
+    }
+  };
+}
+
+export const prompter: Prompter = new AmplifyPrompter();
+
+type Prompter = {
+  confirmContinue: (message?: string) => Promise<boolean>;
+  yesOrNo: (message: string, initial?: boolean) => Promise<boolean>;
+  // options is typed using spread because it's the only way to make it optional if T is a string but required otherwise
+  input: <T = string>(message: string, ...options: T extends string ? [InputOptions<T>?] : [InputOptions<T>]) => Promise<T>;
+  pick: <M extends PickType, T = string>(
+    message: string,
+    choices: Choices<T>,
+    // options is typed using spread because it's the only way to make it required if M is 'many' but optional if M is 'one'
+    ...options: M extends 'one' ? [PickOptions<M>?] : [PickOptions<M>]
+  ) => Promise<PickReturn<M, T>>;
+};
+
+// the following types are the building blocks of the method input types
+
+type HiddenInputOption = {
+  hidden?: boolean;
+};
+
+type InitialSelectionOption<M extends PickType> = {
+  initial?: M extends 'one' ? number : number[];
+};
+
+type InitialValueOption<T> = {
+  initial?: T;
+};
+
+type ValidateValueOption = {
+  validate?: Validator;
+};
+
+type ValidatorCast = (input: string) => string | true | Promise<string> | Promise<true>;
+
+type TransformOption<T> = {
+  transform: (value: string) => T | Promise<T>;
+};
+
+type MaybeTransformOption<T> = T extends string ? Partial<TransformOption<T>> : TransformOption<T>;
+
+type MultiselectOption<M extends PickType> = M extends 'many'
```

Comment:
Should it be `Multiselect` or `MultiSelect`? (Here and elsewhere)

Reply:
looking through enquirer's types / docs, they write it as MultiSelect, so I'll update to that

Reply:
I ended up renaming it to `ReturnSize` and reusing it it the pick and input prompts